doc/en/user/source/security/csp.rst

```diff
@@ -0,0 +1,295 @@
+.. _security_csp:
+
+Content Security Policy
+=======================
+
+The :guilabel:`Content Security Policy` page controls the checks that are performed on GeoServer
+requests that GeoServer will use to set an appropriate Content-Security-Policy (CSP) HTTP
+response header to mitigate cross-site scripting and clickjacking attacks.
+
+.. note::
+    The settings on this page do not affect anything if the CSP header has been disabled using the
+    ``geoserver.csp.shouldSetPolicy`` property or if the ``geoserver.csp.policy`` property is used
+    to set a fixed header value.
+
+.. note::
+    Starting with version 9, Wicket will set its own CSP header with a very strict policy that will
+    overwrite what is set by GeoServer although there is a workaround in place to merge directives
```

Comment:
This is a useful note for me (I was about to ask indeed), but not for an end user, that does not know which version of Wicket we are using. Until the Wicket update branch is merged, we should inform the user about the behavior, based on the current Wicket version.

Suggestion: update this comment, and once it's merged, propose a documentation change in the Wicket upgrade branch.

Reply:
Other than referencing specific Wicket versions, this comment would still be useful to users after the Wicket update. It lets them know that disabling GeoServer's CSP will not disable the CSP on Wicket pages and also explains that the actual CSP header for GeoServer's Wicket pages will be a combination of Wicket's and GeoServer's CSP.

Reply:
The documentation no longer refers to specific Wicket version. The comment may need to be updated if a Wicket upgrade does not happen in the 2.26 release.

Reply:
I removed the comment from the documentation for now and added a note in CSPDefaultConfiguration.java that the default configuration and documentation need to be updated after the Wicket 9 upgrade.